### ISSUE TITLE
Expand `I001` documentation

### ISSUE THREAD

**Issue by r-heimann** [astral-sh/ruff#27232]
### Summary

Hi, after updating to 0.16.0 i noticed that `I001` is failing with any import, even if everything is properly configured. Example (AWS Lambda Python code):

```py
import json

def lambda_handler(event, context):

  test = json("test")
  print(test)
```

gives the error

> Import block is un-sorted or un-formatted
> 
> help: Organize importsRuff[I001](https://docs.astral.sh/ruff/rules/unsorted-imports)

### Version

0.16.0

**Comment by calumy** [comment]
This can be fixed by adding an extra blank line below `import json`. See [playground](https://play.ruff.rs/d74dad7f-eb47-4486-8b5d-c2fe891df975).

**Comment by r-heimann** [comment]
> This can be fixed by adding an extra blank line below `import json`. See [playground](https://play.ruff.rs/d74dad7f-eb47-4486-8b5d-c2fe891df975).

Thank you, i was able to verify that it is fixed - but the Ruff documentation didn't mention anything about spaces? I also noticed that Ruff isn't explaining what exacly is wrong with this configuration:

```py
import json
import logging
import boto3


def lambda_handler(event, context):
    print(boto3, logging, json)
```

After talking to a LLM i saw that 3rd party modules should be separated by an extra space:

```py
import json
import logging

import boto3


def lambda_handler(event, context):
    print(boto3, logging, json)
```

The new rule set doesn't seem to properly explain these kinds of issues.

**Comment by r-heimann** [comment]
I also just noticed that doing this ignores the 2 spaces at the end of the imports:

```py
import json
import logging

import boto3

test = "test"

def lambda_handler(event, context):
    print(boto3, logging, json)
```

Incredibly confusing, but i am not a Python Expert - so i may be stumbling upon something that is already known.

**Comment by ntBre** [comment]
Thanks for the report! I see how this is confusing. As @calumy said, isort and Ruff's `I001` rule require additional blank lines between imports and function and class definitions. The single blank line is allowed between imports and other kinds of statements. You can adjust this with the [lines-after-imports](https://docs.astral.sh/ruff/settings/#lint_isort_lines-after-imports) setting.

Looking at this today, I was a bit surprised that our docs for `I001` are so short. I think it would be a good idea to expand them, especially now that the rule is enabled by default.

**Comment by roian6** [comment]
Thanks! I’d be happy to work on this as a documentation-only change.

I’ll expand the I001 rule documentation to explain import sections and the blank-line requirements after an import block, with links to the relevant `lines-after-imports` and import-classification settings.

**Comment by jonathandung** [comment]
Maybe change the issue title?
